### PR TITLE
perf: use concat! and &'static str for servo version

### PR DESCRIPTION
**Pull request by Integral-Tech**
Replace `format!` and `String` with `concat!` and `&'static str` for servo version.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

### CODE REVIEW

**Review by mrobinson — CHANGES_REQUESTED** [review]
I think for something like this, where the code isn't run in a tight loop, the performance implications aren't very large. It feels a bit like premature optimization to make this change.

File: components/compositing/compositor.rs [hunk 1]
```diff
@@ -364,7 +364,7 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
         exit_after_load: bool,
         convert_mouse_to_touch: bool,
         top_level_browsing_context_id: TopLevelBrowsingContextId,
-        version_string: String,
```
[the remark this answers is not in the document]
Comment (Integral-Tech) [reply components/compositing/compositor.rs:367]: IMO, as the type has been changed to `&'static str`, changing argument / variable name from `version_string` to `version_str` is also preferred.

**Comment by mrobinson** [comment]
Thank you for the contribution. I'd be happy to reconsider it if it's possible to see that it's having a real world effect on performance. For now, I'm not sure we are ready for these kind of micro-optimizations.

**Comment by mrobinson** [comment]
An example of why this might be tricky for embedders is that they do not have the potential to produce a static string at runtime for their version. For instance, if it was read from a configuration file...

**Comment by Integral-Tech** [comment]
@mrobinson The original code uses `env!` macro which gets the environment variable during compilation.

**Comment by Integral-Tech** [comment]
> An example of why this might be tricky for embedders is that they do not have the potential to produce a static string at runtime for their version.

@mrobinson `format!` produces the string at runtime, while `concat!` produces a static string during compilation,

**Pull request closed without merging**